usr/share/rear/prep/default/320_include_uefi_env.sh

```diff
@@ -47,13 +47,21 @@ if grep -qw efivars /proc/mounts; then
     SYSFS_DIR_EFI_VARS=/sys/firmware/efi/efivars
 fi
 
-# next step, is checking /boot/efi directory (we need it)
-if [[ ! -d /boot/efi ]]; then
-    return    # must be mounted
+# next step, is case-sensitive checking /boot for case-insensitive /efi directory (we need it)
+if [[ -n `find /boot -maxdepth 1 -iname efi -type d` ]]; then
+    return    # not found
 fi
 
 # next step, check filesystem partition type (vfat?)
+local efi_mount_point=""
 UEFI_FS_TYPE=$(awk '/\/boot\/efi/ { print $3 }' /proc/mounts)
+# if not mounted at /boot/efi, try /boot itself
+if [[ -z "$UEFI_FS_TYPE" ]]; then
+    UEFI_FS_TYPE=$(awk '/\/boot/ { print $3 }' /proc/mounts)
+    [[ -z "$UEFI_FS_TYHPE" ]] && efi_mount_point='/\/boot/'
```

Comment:
@gozora 
1. Yes, you caught my typo. I will also fix that.
2. The comment "if not mounted at /boot/efi, try /boot ..." tries to explain

The code block is trying to catch an additional UEFI situation, not falling back to ReaR considering the case is legacy boot.

The original code only keeps in mind that `/boot/efi` must be in the mount list. That is not always the case. Having `/efi` as plain subdirectory of a `/boot` mount is still a valid UEFI boot situation.

My UEFI booted Arch Linux: 
```
$ mount | grep vfat
/dev/sdb1 on /boot type vfat (rw,noatime,nodiratime,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
```